src/rate.rs

```diff
@@ -676,21 +677,29 @@ const Q57_SQUARE_EXP_SCALE: f64 =
   (2.0 * ::std::f64::consts::LN_2) / ((1i64 << 57) as f64);
 
 // Daala style log-offset for chroma quantizers
-fn chroma_offset(log_target_q: i64) -> (i64, i64) {
+fn chroma_offset(
+  log_target_q: i64, chroma_sampling: ChromaSampling,
+) -> (i64, i64) {
   let x = log_target_q.max(0);
-  // Gradient 0.266 optimized for CIEDE2000+PSNR on subset3
-  let y = (x >> 2) + (x >> 6);
+  // Gradient optimized for CIEDE2000+PSNR on subset3
+  let y = match chroma_sampling {
+    ChromaSampling::Cs400 => unimplemented!(),
+    ChromaSampling::Cs420 => (x >> 2) + (x >> 6), // 0.266
+    ChromaSampling::Cs422 => (x >> 3) + (x >> 4) - (x >> 7), // 0.180
+    ChromaSampling::Cs444 => (x >> 4) + (x >> 5) + (x >> 8), // 0.098
```

Comment:
I think you should have a 1.0 entry for when the colorspace is RGB and the subsampling is 444.

Reply:
That makes sense and we should plumb `ColorDescription` next. In this change, we are still presuming BT.709 but no longer assuming 4:2:0 chroma quantization behavior. Ultimately, we should map all color configurations to optimal chroma quantizer offsets.